Sources/Sentry/include/SentryClient.h

```diff
@@ -45,77 +47,20 @@ SENTRY_NO_INIT
 
 @property(nonatomic, strong) SentryOptions *options;
 
-/**
- * This property will be filled before the event is sent.
- */
-@property(nonatomic, copy) NSString *_Nullable releaseName;
 
 /**
- * This property will be filled before the event is sent.
+ * This block can be used to modify the event before it will be serialized and sent
  */
-@property(nonatomic, copy) NSString *_Nullable dist;
+@property(nonatomic, copy) SentryBeforeSerializeEvent _Nullable beforeSerializeEvent;
 
-/**
- * The environment used for this event
- */
-@property(nonatomic, copy) NSString *_Nullable environment;
-    
 /**
  * This will be filled on every startup with a dictionary with extra, tags, user which will be used
  * when sending the crashreport
  */
 @property(nonatomic, strong) NSDictionary<NSString *, id> *_Nullable lastContext;
 
-/**
- * Contains the last successfully sent event
- */
-@property(nonatomic, strong) SentryEvent *_Nullable lastEvent;
-    
-/**
- * Is the client enabled?. Default is @YES, if set @NO sending of events will be prevented.
- */
-@property(nonatomic, copy) NSNumber *enabled;
+@property(nonatomic, strong) SentryTransport* transport;
```

Comment:
Property shouldn't be public, people should just be able to set it through the options.

Reply:
true. But what to do with `SentryBeforeSendRequest`? I would put it in options if that makes sense to you.

Reply:
Yes, put it in the options and just call it `SentryBeforeSend`